core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -799,7 +801,14 @@ public ServerBuilder withRoute(Consumer<ServiceBindingBuilder> customizer) {
      * Returns a {@link ServiceBindingBuilder} which is for binding a {@link Service} fluently.
      */
     public ServiceBindingBuilder route() {
-        return new ServiceBindingBuilder(this);
+        return new ServiceBindingBuilder(this, RouteBindingBuilder.ofOnePath());
+    }
+
+    /**
+     * Returns a {@link DecoratingServiceBindingBuilder} which is for binding a {@code decorator} fluently.
+     */
+    public DecoratingServiceBindingBuilder decoratingRoute() {
```

Comment:
Maybe intead `routeDecorator`? I think this registers a decorator for a set of routes, not a route itself.

Reply:
Also maybe just `decorator()` and deprecate the existing methods, which would be `catchAll` route decorators. I think something like that would make it a bit more clear that a decorator registered like that only applies to routed requests (currently `ServerBuilder.decorator` feels like it should apply to all requests, even 404 #1625).

Reply:
`decorator()` Sounds good to me.

Reply:
Hmm.. at second thought, I feel like we need something similar to `service()` and `serviceUnder()` for decorators, i.e.

- `decoratingRoute()` -> `routeDecorator()`
- Add `decorator(String path, ...)`
- Add `decoratorUnder(String pathPrefix, ...)`
- Add `decorator(Route route, ...)`
  - I guess we need this as well because:
    - There's also `service(Route, ...)`.
    - A user might want to reuse his or her `Route` instance.